src/coreclr/src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -7030,7 +7032,184 @@ HRESULT ProfToEEInterfaceImpl::SetEnvironmentVariable(const WCHAR *szName, const
+                  && offsetof(EventPipeProviderConfiguration, m_keywords) == offsetof(COR_PRF_EVENTPIPE_PROVIDER_CONFIG, keywords)
+                  && offsetof(EventPipeProviderConfiguration, m_loggingLevel) == offsetof(COR_PRF_EVENTPIPE_PROVIDER_CONFIG, loggingLevel)
+                  && offsetof(EventPipeProviderConfiguration, m_pFilterData) == offsetof(COR_PRF_EVENTPIPE_PROVIDER_CONFIG, filterData)
+                  && sizeof(EventPipeProviderConfiguration) == sizeof(COR_PRF_EVENTPIPE_PROVIDER_CONFIG),
+        "Layouts of EventPipeProviderConfiguration type and COR_PRF_EVENTPIPE_PROVIDER_CONFIG type do not match!");
+
+    if (szProviderName == NULL
+        || cProviderConfigs == 0
+        || pProviderConfigs == NULL
+        || pSession == NULL)
+    {
+        return E_INVALIDARG;
+    }
+
+    HRESULT hr = S_OK;
+    EX_TRY
+    {
+        auto eventPipeFunc = [](EventPipeProvider *provider,
+                                DWORD eventId,
+                                DWORD eventVersion,
+                                ULONG cbMetadataBlob,
+                                LPCBYTE metadataBlob,
+                                ULONG cbEventData,
+                                LPCBYTE eventData,
+                                LPCGUID pActivityId,
+                                LPCGUID pRelatedActivityId,
+                                Thread *pEventThread,
+                                ULONG numStackFrames,
+                                UINT_PTR stackFrames[])
+        {
+            BEGIN_PIN_PROFILER(CORProfilerIsMonitoringEventPipe());
+            g_profControlBlock.pProfInterface->EventPipeEventDelivered(provider,
+                                                                       eventId,
+                                                                       eventVersion,
+                                                                       cbMetadataBlob,
+                                                                       metadataBlob,
+                                                                       cbEventData,
+                                                                       eventData,
+                                                                       pActivityId,
+                                                                       pRelatedActivityId,
+                                                                       pEventThread,
+                                                                       numStackFrames,
+                                                                       stackFrames);
+            END_PIN_PROFILER();
+        };
+
+        EventPipeProviderConfiguration *pProviders = reinterpret_cast<EventPipeProviderConfiguration *>(pProviderConfigs);
+        UINT64 sessionID = EventPipe::Enable(NULL,
+                                             100, // circularBufferSizeInMB (shouldn't need this once it's synchronous)
```

Comment:
nit: is there anything needed to rectify this comment?

Reply:
I have the same question. The idea is the profiler api delivery is more efficient than the async path. Is that not correct?

Reply:
It already was synchronous, but this comment made me realize we were still creating the buffer. I just pushed changes so now when using profiler we never even create the buffer.